prov/sockets/include/sock.h

```diff
@@ -1211,6 +1218,14 @@ int sock_epoll_wait(struct sock_epoll_set *set, int timeout);
 int sock_epoll_get_fd_at_index(struct sock_epoll_set *set, int index);
 void sock_epoll_close(struct sock_epoll_set *set);
 
+int sock_epoll_map_init(struct sock_epoll_map *emap);
+int sock_epoll_map_destroy(struct sock_epoll_map *emap);
+int sock_epoll_map_add(struct sock_epoll_map *emap, int fd, void *ptr);
+int sock_epoll_map_del(struct sock_epoll_map *emap, int fd);
+void *sock_epoll_map_lookup(struct sock_epoll_map *emap, int fd);
+int sock_epoll_map_signal(struct sock_epoll_map *emap);
+int sock_epoll_map_wait(struct sock_epoll_map *emap);
```

Comment:
I started a patch that converted sockets to use fi_epoll calls.  It was straightforward, except for the use in sock_pe_progress_rx_ep().  I couldn't follow what that call was doing exactly, or figure out if it was correct.  The way it's written, it looks like it could fail to progress a connection.  I don't have time to look into this in more detail at the moment.

Note that the fi_epoll_wait abstraction currently only returns information about the first event.  That functionality could be expanded to return multiple events if it is needed.